test/adapters/adapters.spec.js

```diff
@@ -0,0 +1,195 @@
+        interval: 1000
+    });
+}
+
+describe('adapters', () => {
+
+    before(() => {
+        var tmpdir = tmp.dirSync().name;
+        return Promise.all([
+            dockerUtils.destroyContainer('elasticsearch-adapter-test'),
+            dockerUtils.destroyContainer('influxdb-adapter-test'),
+            dockerUtils.destroyContainer('graphite-adapter-test'),
+            dockerUtils.destroyContainer('juttle-engine-local')
+        ])
+        .then(() => {
+            return dockerUtils.checkJuttleEngineLocalExists();
+        })
+        .then(() => {
+            return Promise.all([
+                dockerUtils.runContainer({
+                    name: 'elasticsearch-adapter-test',
+                    image: 'elasticsearch:1.5.2',
+                    detach: true
+                }),
+                dockerUtils.runContainer({
+                    name: 'influxdb-adapter-test',
+                    image: 'tutum/influxdb:0.10',
+                    detach: true
+                }),
+                dockerUtils.runContainer({
+                    name: 'graphite-adapter-test',
+                    image: 'sitespeedio/graphite:0.9.14',
+                    detach: true
+                })
+            ]);
+        })
+        .then(() => {
+            var config = JSON.stringify({
+                adapters: {
+                    elastic: {
+                        address: 'elasticsearch',
+                        port: 9200
+                    },
+                    influx: {
+                        url: 'http://influxdb:8086'
+                    },
+                    graphite: {
+                        carbon: {
+                            host: 'graphite',
+                            port: 2003
+                        },
+                        webapp: {
+                            host: 'graphite',
+                            port: 80,
+                            username: 'guest',
+                            password: 'guest'
+                        }
+                    }
+                }
+            });
+
+            var filename = path.join(tmpdir, '.juttle-config.json')
+            return fs.writeFileAsync(filename, config)
+        })
+        .then(() => {
+            return dockerUtils.runContainer({
+                name: 'juttle-engine-local',
+                image: 'juttle-engine:local',
+                ports: ['8080:8080'],
+                links: [
+                    'elasticsearch-adapter-test:elasticsearch',
+                    'influxdb-adapter-test:influxdb',
+                    'graphite-adapter-test:graphite'
+                ],
+                volumes: [`${tmpdir}:/tmp`],
+                workdir: '/tmp',
+                detach: true
+            });
+        })
+        .then(() => {
+            return waitForSuccess('elasticsearch-adapter-test',
+                                  ['curl', 'http://localhost:9200']);
+        });
+    });
+
+    after(() => {
+        /*
```

Comment:
fixed and thanks as that would let containers around running for no good reason.